azkaban-common/src/test/java/azkaban/executor/container/KubernetesContainerizedImplTest.java

```diff
@@ -259,10 +262,11 @@ public void testPodConstruction() throws Exception {
     allImageTypes.add(AZKABAN_BASE_IMAGE);
     allImageTypes.add(AZKABAN_CONFIG);
     allImageTypes.addAll(jobTypes);
+    allImageTypes.addAll(dependencySet);
     final VersionSet versionSet = this.kubernetesContainerizedImpl
         .fetchVersionSet(flow.getExecutionId(), flowParam, allImageTypes, flow);
     final V1PodSpec podSpec = this.kubernetesContainerizedImpl
-        .createPodSpec(flow.getExecutionId(), versionSet, jobTypes, flowParam);
+        .createPodSpec(flow.getExecutionId(), versionSet, jobTypes, dependencySet, flowParam);
```

Comment:
Is there a corresponding yaml output for the pod-spec that needs to be updated for this ?

Reply:
Create PodSpec will take into account creating the PodSpec object including dependencySet/dependencyTypes. This object is used to create POD which is used for dispatching to K8s. We end up logging the POD yaml output if dev pod is enabled.